### DISCUSSION TITLE
when "--ip6tables enables IPv6 iptables rules (only if experimental)" will be non-experimental?

### DISCUSSION THREAD

**Discussion by WallaceSUI** [moby/moby#45384] [General]
Hey team,

Sorry to trouble you. I knew that previously you mentioned ip6tables configuration through https://github.com/moby/moby/pull/41622 and it is experimental only. But I wonder when (by date or by release) you plan to make this implementation as non-experimental (remove parameter "experimental")? I didn't see any changes in the latest docker version.

Thanks a lot.

**Comment by akerouanton** [comment]
Hi @WallaceSUI, this is still something we have to figure out. We have a lot to do before IPv6 is really a first-class citizen. Nonetheless, the `ip6tables` config param is probably not so much of an experimental thing now. I'll put this on my todo list and discuss it with other maintainers soon.

**Reply by polarathene** [reply]
It would be good if `experimental` can be dropped. I'm not sure if that is tied to any other features or causes any other unrelated feature changes beyond enabling `ip6tables`.

The IPv6 docs could be improved too, and not having this extra step is helpful. I'm currently revising IPv6 config docs for a docker focused project since our users often have trouble with official advice and the guidance from third-party resources over time isn't necessary correct or missing some information :sweat_smile: 

I think `ip6tables` can remain opt-in, and the docs can still express that support for IPv6 itself is still being worked on? Private NAT6 network with ULA subnet seems to be fine and familiar for users though for the most part. That can perhaps be endorsed via docs. GUA subnet is a bit more fragile/variable and affected by the host environment.

**Reply by akerouanton** [reply]
> It would be good if experimental can be dropped. I'm not sure if that is tied to any other features or causes any other unrelated feature changes beyond enabling ip6tables.

We have since introduced feature flags. I don't think we'll add one for ip6tables, it'd be best to move it out of experimental state.

> The IPv6 docs could be improved too

Yeah, the docs page about IPv6 is pretty much all wrong, I need to fix that. It's part of my todo list but I need to work on some other subjects first.
